components/cd-dropdown/cd-dropdown.js

```diff
@@ -356,10 +354,26 @@
       // Add toggling function.
       toggler.addEventListener('click', this.handleToggle);
 
-      // Collapse when pressing scape.
+      // Collapse when pressing escape.
       toggler.addEventListener('keydown', this.handleEscape);
       element.addEventListener('keydown', this.handleEscape);
 
+      // Hide the menu when the focus moves from inside the menu to
+      // an element outside the menu or its toggler.
+      element.addEventListener('focusout', (ev) => {
+        if (ev.relatedTarget && !element.contains(ev.relatedTarget) && toggler !== ev.relatedTarget) {
+          this.toggle(toggler, true);
+        }
+      });
+
+      // Hide the menu when the focus moves from the toggler to
+      // an element outside the menu.
+      toggler.addEventListener('focusout', (ev) => {
+        if (ev.relatedTarget && !element.contains(ev.relatedTarget) && element !== ev.relatedTarget) {
+          this.toggle(toggler, true);
+        }
+      });
+
```

Comment:
```suggestion
      // Hide the dropdown menu when focus changes.
      // Do not apply when the menu is not a dropdown which is currently the
      // case, in practice, when using the 'data-cd-insert-after' attribute.
      // @todo remove when we introduce the Disclosure component.
      if (!element.hasAttribute('data-cd-insert-after')) {
        // Hide the menu when the focus moves from inside the menu to
        // an element outside the menu or its toggler.
        element.addEventListener('focusout', (ev) => {
          if (ev.relatedTarget && !element.contains(ev.relatedTarget) && toggler !== ev.relatedTarget) {
            this.toggle(toggler, true);
          }
        });

        // Hide the menu when the focus moves from the toggler to
        // an element outside the menu.
        toggler.addEventListener('focusout', (ev) => {
          if (ev.relatedTarget && !element.contains(ev.relatedTarget) && element !== ev.relatedTarget) {
            this.toggle(toggler, true);
          }
        });
      }
```

@rupl The suggestion just wraps the event listeners in a condition that ensures that they are not set for `disclosure` cases so that there is no breaking change, leaving us more time to create the disclosure component.